packages/electron-app/src/main/index.ts

```diff
@@ -4,21 +4,72 @@
 
 import electron from 'electron';
 import installExtension, { REACT_DEVELOPER_TOOLS } from 'electron-devtools-installer';
-import Pino from 'pino';
 import path from 'path';
+import Pino from 'pino';
 import url from 'url';
 
-import { CSP, staticPath } from './util';
+import { CSP, isSubstrateRunning, runSubstrateDev, staticPath } from './util';
 
 const { app, BrowserWindow, session } = electron;
-const pino = Pino();
-let mainWindow: Electron.BrowserWindow | undefined;
+const pino = new Pino();
+let sluiApp: Electron.BrowserWindow | undefined;
+let substratePid: number;
+let hasCalledInitParitySubstrate = false;
+
+pino.info('Platform detected: ', process.platform);
+pino.info('Process type: ', process.type);
+pino.info('Process ID: ', process.pid);
+pino.info('Process args: ', process.argv);
+pino.info('Electron version: ', process.versions['electron']);
+
+app.once('ready', async () => {
+  const setPid = (pid: number) => {
+    pino.info('setting substrate pid -> ', pid);
+    substratePid = pid;
+  };
+
+  // https://electronjs.org/docs/tutorial/security#electron-security-warnings
+  process.env.ELECTRON_ENABLE_SECURITY_WARNINGS = 'true';
+
+  if (await isSubstrateRunning()) {
+    pino.error('Substrate instance is allready running!');
+    console.error('Substrate instance is allready running!');
+    // do nothing
+    pino.info('Substrate is running, do nothing.');
+    return;
+  } else if (hasCalledInitParitySubstrate) {
+    pino.error('Unable to initialise Parity Substrate more than once');
+    return;
+  } else {
+    runSubstrateDev(setPid);
+    pino.info('Running Parity Substrate');
+    hasCalledInitParitySubstrate = true;
+  }
+
+  createWindow();
+});
+
+// Quit when all windows are closed.
+app.on('window-all-closed', function () {
+  // On OS X it is common for applications and their menu bar
+  // to stay active until the user quits explicitly with Cmd + Q
+  if (process.platform !== 'darwin') {
+    app.quit();
+  }
 
-// https://electronjs.org/docs/tutorial/security#electron-security-warnings
-process.env.ELECTRON_ENABLE_SECURITY_WARNINGS = 'true';
+  process.kill(substratePid);
```

Comment:
This kills the node when we close the window, which is not what we want. Have a look at https://github.com/paritytech/fether/blob/master/packages/fether-electron/src/main/index.js#L87-L103 (we don't need the lock thing).